src/lib/cozy-client/slices/sharings.js

```diff
@@ -424,8 +424,35 @@ const getSharingRecipients = (state, sharings) =>
     )
     .reduce((a, b) => a.concat(b), [])
 
-const buildSharingLink = (id, doctype, sharecode) =>
-  `${window.location.origin}/public?sharecode=${sharecode}&id=${id}${
+const getAppForDoctype = doctype => {
+  switch (doctype) {
+    case 'io.cozy.files':
+      return 'drive'
+    case 'io.cozy.photos.albums':
+      return 'photos'
+    default:
+      throw new Error(
+        `Sharing link: don't know which app to use for doctype ${doctype}`
+      )
+  }
+}
+
+const getAppUrl = (state, doctype) => {
+  // TODO: here we access a slice of state we don't "own" and that's a pb
+  // we should store the cozy url in the store ourselves
+  if (state.mobile && state.mobile.settings) {
+    const host = state.mobile.settings.serverUrl
+    const app = getAppForDoctype(doctype)
+    return host.replace(
+      /^(https?:\/\/)(\w+)\.(\w+)\.(\w+)/g,
```

Comment:
I think this regex is fragile, because it assumes the server url follows the form `https://{name}.{provider}.{tld and the rest}`. While this is true for the cozy we provide, it's may not always be like this, especially for self-hosted users. But I don't know enough about how the apps URL are handled, maybe it would be safer to check this with the back-end team.

Reply:
If I serve cozy-stack directly on the main domain of mine: `http://ticabri-cozy.com` it breaks.